firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -246,27 +213,113 @@ class Ssp final : public SspInterface, protected Lpc40xxSystemController
     }
     return static_cast<uint16_t>(ssp_registers[pssp]->DR);
   }
-  // Sets the various modes for the Peripheral
-  // @param mode - master or slave mode
-  // @param frame - format for Peripheral data to use
-  // @param size - number of bits per frame
-  void SetPeripheralMode(MasterSlaveMode mode, FrameMode frame,
-                         DataSize size) override
+
+  /// Sets up SSP peripheral as SPI master
+  void SetSpiMasterDefault() override

```

Comment:
SetSpiMasterDefault shouldn't be doing any bit manipulation. As you can see, you are doing double duty here and writing the same set of code twice. You can simply call/reuse the methods you have already made rather than doing the direct register manipluation.